### ISSUE TITLE
New Rule: require-newline

### ISSUE THREAD

**Issue by BHyeonKim** [eslint/markdown#714]
### Rule details

This rule requires a newline(`\n`) at the end of the markdown file. On Unix systems, text files are expected to end with a newline — otherwise, they're not considered valid text files. 

### What type of rule is this?

Warns about a potential problem

### Example code

```markdown
# Heading

This should trigger warning.
```

### Prior Art

[markdownlint-md047](https://github.com/DavidAnson/markdownlint/blob/main/doc/md047.md)

### Participation

- [x] I am willing to submit a pull request to implement this rule.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

### Additional comments
Disclosure: I'm a participant of [open source contribution program OSSCA](https://github.com/eslint-ossca)

_No response_

**Comment by lumirlumir** [comment]
> Disclosure: I'm a participant of [open source contribution program OSSCA](https://github.com/eslint-ossca): confirmed.

Thank you for opening an issue. However, under our policy, we don’t accept stylistic rules in ESLint core, and existing stylistic rules are feature-frozen.

Please read our policy on this here: https://eslint.org/docs/latest/use/core-concepts/glossary#stylistic-rule 

Thank you for understanding.
